synapse/webhooks.py

```
@@ -0,0 +1,297 @@
+    def register(
+        self,
+        url: str,
+        events: list[str] | None = None,
+        secret: str | None = None,
+        metadata: dict[str, Any] | None = None
+    ) -> WebhookConfig:
+        """Register a new webhook."""
+        # Validate URL
+        parsed = urlparse(url)
+        if not parsed.scheme or not parsed.netloc:
+            raise ValueError(f"Invalid webhook URL: {url}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)